src/domains/citizen-auth-app/10_function_public.tf

```diff
@@ -0,0 +1,258 @@
+}
+
+module "function_public_itn" {
+  source = "git::https://github.com/pagopa/terraform-azurerm-v3.git//function_app?ref=v8.44.0"
+
+  resource_group_name = azurerm_resource_group.public_rg_itn.name
+  name                = format("%s-public-func-01", local.short_project_itn)
+  location            = local.itn_location
+  domain              = "auth"
+  health_check_path   = "/info"
+
+  node_version    = "18"
+  runtime_version = "~4"
+
+  app_service_plan_info = {
+    kind                         = var.function_public_kind
+    sku_size                     = var.function_public_sku_size
+    maximum_elastic_worker_count = 0
+    worker_count                 = null
+    zone_balancing_enabled       = true
+  }
+
+  always_on                                = "true"
+  application_insights_instrumentation_key = data.azurerm_application_insights.application_insights.instrumentation_key
+
+  app_settings = merge(
+    local.function_public.app_settings_common,
+  )
+
+  internal_storage = {
+    "enable"                     = false,
+    "private_endpoint_subnet_id" = data.azurerm_subnet.itn_pep.id,
+    "private_dns_zone_blob_ids"  = [data.azurerm_private_dns_zone.privatelink_blob_core_windows_net.id],
+    "private_dns_zone_queue_ids" = [data.azurerm_private_dns_zone.privatelink_queue_core_windows_net.id],
+    "private_dns_zone_table_ids" = [data.azurerm_private_dns_zone.privatelink_table_core_windows_net.id],
+    "queues"                     = [],
+    "containers"                 = [],
+    "blobs_retention_days"       = 0,
+  }
+
+  subnet_id = module.function_public_snet_itn.id
+
+  allowed_subnets = [
+    module.function_public_snet_itn.id,
+    data.azurerm_subnet.apim_v2_snet.id,
+  ]
+
+  # Action groups for alerts
+  action = [
+    {
+      action_group_id    = data.azurerm_monitor_action_group.error_action_group.id
+      webhook_properties = {}
+    }
+  ]
+
+  tags = var.tags
+}
+
+module "function_public_staging_slot_itn" {
+  source = "git::https://github.com/pagopa/terraform-azurerm-v3.git//function_app_slot?ref=v8.44.0"
+
+  name                = "staging"
+  location            = local.itn_location
+  resource_group_name = azurerm_resource_group.public_rg_itn.name
+  function_app_id     = module.function_public_itn.id
+  app_service_plan_id = module.function_public_itn.app_service_plan_id
+  health_check_path   = "/info"
+
+  storage_account_name       = module.function_public_itn.storage_account.name
+  storage_account_access_key = module.function_public_itn.storage_account.primary_access_key
+
+  node_version                             = "18"
+  always_on                                = "true"
+  runtime_version                          = "~4"
+  application_insights_instrumentation_key = data.azurerm_application_insights.application_insights.instrumentation_key
+
+  app_settings = merge(
+    local.function_public.app_settings_common,
+  )
+
+  subnet_id = module.function_public_snet_itn.id
+
+  allowed_subnets = [
```

Comment:
Same as above

Reply:
I removed the unnecessary in https://github.com/pagopa/io-infra/pull/1244/commits/8feb975ccc66152b87932e21eb9a2ef89b5ee2eb